aptos-move/framework/src/module_metadata.rs

```diff
@@ -146,8 +146,35 @@ pub fn get_vm_metadata_v0(vm: &MoveVM, module_id: ModuleId) -> Option<RuntimeMod
     }
 }
 
+/// Check if the metadata has unknown key/data types
+pub fn check_metadata_format(module: &CompiledModule) -> Result<(), MalformedError> {
```

Comment:
I think what we also would need here is feature gating. Say we support KEY_v1 and KEY_v2 in the code, but we haven't enabled KEY_v2 yet, as it is feature gated. We would like that KEY_V2 is rejected. Perhaps we can pass a predicate `enabled_key` to express this, and the caller side decides this? One SEV we had is where a new key was landed on chain before it was actually enabled, this would prevent this.

Reply:
yes, but isn't metadata v1 already enabled? for next new key we definitely we should feature gate it

Reply:
Yes its all on predicate is `|_| true` but you could add the mechanism now so its correctly used in the future. But fine also for another PR.